sources/web/datalab/polymer/components/file-browser/file-browser.ts

```diff
@@ -721,7 +737,7 @@ class FileBrowserElement extends Polymer.Element implements DatalabPageElement {
     // Wait on all upload requests before declaring success or failure.
     try {
       await Promise.all(uploadPromises);
-      this._fetching = false;
+      this._busy = false;
```

Comment:
Given that you are allowing a second operation to start when a first is running, I think it's going to be pretty easy to get this busy-flag out of sync: start an async operation, then start another one just before the first one finishes. The busy-flag will get cleared when the first one finishes, and the second one will still be running, but the busy-flag will not be set. I'm fine with leaving this for another PR, since I think the current code is already problematic, but perhaps putting in a TODO at one of these places would be appropriate. Unless I misunderstand this flag, of course.

Reply:
Your understanding is correct, I guess we could use the counter mechanism here. I'll put a TODO for now.